tests/factories/oauth2_token.py

```diff
@@ -11,5 +13,6 @@
     user_id=USER_ID,
     application_instance=SubFactory(ApplicationInstance),
     access_token=ACCESS_TOKEN,
+    received_at=datetime(2023, 12, 1),
```

Comment:
Yes. Factories should normally use "random" data, this has a couple of benefits:

* It removes irrelevant values from tests. For example if a test wants a token and all it cares about is the token's `user_id`, and the other attributes of the token are irrelevant to the test, the tests can just do `factories.OAuth2Token(user_id="...")` and let the rest of the values be randomly generated. Rather than having to do `models.OAuth2Token(user_id="...", application_instance_id=..., access_token=..., ...)` when `user_id` is the only relevant attribute. This removes a lot of distracting noise from test code and brings out the important values.

* Conversely, and more to the point here, another benefit of factories using random data by default is that it _forces_ tests to include values in the test code when those values are important to the test. If your test needs a token for a particular user you _have_ to specify the `user_id` to `factories.OAuth2Token(user_id="...")`. Otherwise the factory will generate a random user for you. This is better than the factory always creating a token for a hard-coded user `bob` and then tests invisibly relying on that fact and just doing `factories.OAuth2Token()` and having the test depend on the hidden fact that the factory always creates a token for the user `bob`, which won't be obvious to anyone just reading the test code.

  In your case, if you have tests that are reliant on the fact that the tokens being generated are old, this _should_ be explicit in your test code by having the tests pass in an old `received_at` date to the factory.